### PR TITLE
Add release notes guideline to contributors guide

### PR DESCRIPTION
**Pull request by saschagrunert**
Hey :wave:, I'd like to push forward #484 and think a good first step would be to put the information into the contributors guide. After this we should discuss how we apply the guide to be more visible to all users.

Refers to: #484

### CODE REVIEW

File: contributors/guide/release-notes.md [hunk 1]
```diff
@@ -4,7 +4,48 @@ weight: 1
 slug: "release-notes"
 ---
 
-On the kubernetes/kubernetes repository, release notes are required for any pull request with user-visible changes, such as bug-fixes, feature additions, and output format changes.
+On the kubernetes/kubernetes repository, release notes are required for any pull
+request with user-visible changes, such as bug-fixes, feature additions, and
```
Comment (idvoretskyi) [inline contributors/guide/release-notes.md]: Nit: "bug fixes" or "bugfixes", not "bug-fixes".

File: contributors/guide/release-notes.md [hunk 2]
```diff
@@ -4,7 +4,48 @@ weight: 1
...
+- build infrastructure
+- fixes of bugs which have not been released
+- fixes of minor bugs
+- incomplete or inaccessible features
+
+## Content of a Release Note
+
+A release note needs a clear, concise description of the change. This includes:
+
+1. an indicator if it was _Added_, _Changed_, _Fixed_, _Removed_, _Deprecated_
```
Comment (justaugustus) [inline contributors/guide/release-notes.md]:
```suggestion
1. an indicator if the pull request _Added_, _Changed_, _Fixed_, _Removed_, _Deprecated_ functionality
```

File: contributors/guide/release-notes.md [hunk 3]
```diff
@@ -4,7 +4,48 @@ weight: 1
...
+   or changed in terms of maturity (alpha, beta, stable)
```
Comment (justaugustus) [inline contributors/guide/release-notes.md]:
```suggestion
   or changed enhancement/feature maturity (alpha, beta, stable)
```

File: contributors/guide/release-notes.md [hunk 4]
```diff
@@ -4,7 +4,48 @@ weight: 1
...
+2. an indicator if there is user _Action required_
+3. the name of the affected API or configuration fields, CLI commands/flags or
+   feature
```
Comment (justaugustus) [inline contributors/guide/release-notes.md]:
```suggestion
   enhancement/feature
```

File: contributors/guide/release-notes.md [hunk 5]
```diff
@@ -4,7 +4,48 @@ weight: 1
...
+4. a link to relevant user documentation about the feature
```
Comment (justaugustus) [inline contributors/guide/release-notes.md]:
```suggestion
4. a link to relevant user documentation about the enhancement/feature
```

**Comment by saschagrunert** [comment]
> Nice work, @saschagrunert!
> I've got a few nits to address.

Thanks for reviewing, I updated it once again.

**Comment by castrojo** [comment]
/approve

Nice work!

**Pull request merged**